scripts/codemods/typed_flags.py

```diff
@@ -0,0 +1,186 @@
+        # thankfully, we aren't limited to just codemodding, we can import the module if we need it
+        # if __init__ exists without a TYPE_CHECKING block, we create and add two overloads
+        # if __init__ exists under a TYPE_CHECKING block, we remove and replace it
+        # we also check and remove previously existing TYPE_CHECKING and overloads
+
+        # and we have to import the module anyways, to get a list of valid flags for each subclass.
+        flag: flags.BaseFlags = getattr(self.imported_module, node.name.value)
+        all_flags = sorted(flag.VALID_FLAGS.keys())
+
+        # determine which method to use for creating the init overload, and possibly the update overload too.
+        # only a few of these need to be overloads, in which we can use the same technique from permission overloads.
+        # for the other ones, we need to find a TYPE_CHECKING block that *may* or may not contain a function definition.
+        # if none of the TYPE_CHECKING blocks contain a function definition, we make a new one and
+        # insert it near the beginning of the class body.
+        # we also decorate with @_generated so we can delete it later.
+
+        if_block: Optional[cst.If] = None
+        init: Optional[cst.FunctionDef] = None
+        body = list(node.body.body)
+        kwonly_params = [
+            cst.Param(cst.Name(flag_name), cst.Annotation(cst.Name("bool")), default=cst.Ellipsis())
+            for flag_name in all_flags
+        ]
+        for b in body.copy():
+            if m.matches(b, m.FunctionDef(m.Name("__init__"))) and isinstance(b, cst.FunctionDef):
+                if any(m.matches(deco.decorator, m.Name("_generated")) for deco in b.decorators):
+                    body.remove(b)
+                    continue
+                init = b
+                break
+
+        if not init:
+
+            # find the existing one if one exists
+            for b in body:
+                if m.matches(b, m.If(test=m.Name("TYPE_CHECKING"))):
+                    # iterate through the options to see if one of the body is __init__
+                    for line in b.body.body:  # type: ignore
+                        if m.matches(line, m.FunctionDef(m.Name("__init__"))):
+                            if_block = b  # type: ignore
+                            break
+                if if_block:
+                    break
+            else:
+
+                # one doesn't currently exist, great.
+                # so we have two options here... we could make a typechecking block from scratch or... we could cheat
+                code = textwrap.dedent(
+                    """
+                    if TYPE_CHECKING:
+                        @_generated
+                        def __init__(self):
+                            ...
+                    """
+                )
+                if_block = cast("cst.If", cst.parse_statement(code))
```

Comment:
```suggestion
                if_block = cast("cst.If", cst.parse_statement(code, config=self.module.config_for_parsing))
```
https://libcst.readthedocs.io/en/latest/best_practices.html#provide-a-config-when-generating-code-from-templates

Reply:
https://github.com/DisnakeDev/disnake/pull/778/commits/469daa83a02e7dc88f07702b6ce5bb57f6a7de70